sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/AssemblyReferenceService.cs

```diff
@@ -0,0 +1,102 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+using NuGet;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.UnitTest
+{
+    internal static class AssemblyReferenceService
+    {
+        private const string PackagesFolderRelativePath = @"..\..\..\..\packages\";
+
+        private static readonly string systemAssembliesFolder =
+            new FileInfo(typeof(object).Assembly.Location).Directory.FullName;
+
+        private static readonly PackageManager packageManager =
+            new PackageManager(CreatePackageRepository(), PackagesFolderRelativePath);
+
+        private static readonly Dictionary<AssemblyReference, MetadataReference> cache =
+            new Dictionary<AssemblyReference, MetadataReference>();
+
+        private static readonly List<string> allowedNugetLibDirectories =
```

Comment:
Dlls in folder `netstandard1.3`+ results in failures (most likely because of the current fwk version).